app/Models/Characters/Monster.php

```diff
@@ -31,6 +32,38 @@ public function skills()
         return $this->belongsToMany(Skill::class)->withPivot('bonus');
     }
 
+    public function items()
+    {
+        return $this->belongsToMany(Item::class);
+    }
+
+    public function getArmorAttribute()
+    {
+        return $this->items()
+            ->whereHas('type', function ($type) {
+                $type->where('code', 'armor');
```

Comment:
Dot notation doesn't work does it?
like
->where('type.code', 'armour');

Reply:
Well thats unfortunate. You can't with the query where but you can with the collection where.

Reply:
If you look at whereHas in Laravel it looks like it should work...:

It checks if relation is dot notated, and therefore nested?
```php
<?php

namespace Illuminate\Database\Eloquent\Concerns\QueriesRelationships;

...

public function has($relation, $operator = '>=', $count = 1, $boolean = 'and', Closure $callback = null)
{
    if (is_string($relation)) {
        if (strpos($relation, '.') !== false) {
            return $this->hasNested($relation, $operator, $count, $boolean, $callback);
        }

        $relation = $this->getRelationWithoutConstraints($relation);
    }
...
```

Reply:
Yes it works for whereHas but not where